lib/Alchemy/Phrasea/SearchEngine/Elastic/Indexer/Record/Hydrator/SubDefinitionHydrator.php

```diff
@@ -11,76 +11,66 @@
 
 namespace Alchemy\Phrasea\SearchEngine\Elastic\Indexer\Record\Hydrator;
 
+use Alchemy\Phrasea\Application;
 use databox;
 use Doctrine\DBAL\Connection;
+use media_Permalink_Adapter;
 
 class SubDefinitionHydrator implements HydratorInterface
 {
+    /** @var Application  */
+    private $app;
+
     /** @var databox */
     private $databox;
 
-    public function __construct(databox $databox)
+    /** @var  boolean */
+    private $populatePermalinks;
+
+    public function __construct(Application $app, databox $databox, $populatePermalinks)
     {
+        $this->app = $app;
         $this->databox = $databox;
+        $this->populatePermalinks = $populatePermalinks;
     }
 
     public function hydrateRecords(array &$records)
     {
-        $sql = <<<SQL
-            SELECT
-              s.record_id,
-              s.name,
-              s.height,
-              s.width,
-              CONCAT(TRIM(TRAILING '/' FROM s.path), '/', s.file) AS path
-            FROM subdef s
-            WHERE s.record_id IN (?)
-            ORDER BY s.record_id
-SQL;
-        $statement = $this->databox->get_connection()->executeQuery($sql,
-            array(array_keys($records)),
-            array(Connection::PARAM_INT_ARRAY)
-        );
+        foreach(array_keys($records) as $rid) {
+            try {
+                $subdefs = $this->databox->getRecordRepository()->find($rid)->get_subdefs();
```

Comment:
this is the sloooow point.
We should include this into the "if", but that means restoring most of the previous code (sql) into a big "else".
So for readability, for now the best is : 

```
pub fct hydrateRecords(...)
{
  if(test) hydrateRecordsWithPermalinks(...) else hydrateRecordsWithoutPermalink(...)
}
priv hydrateRecordsWithPermalinks(...) { new code };
priv hydrateRecordsWithoutPermalinks(...) { old code };
```